app/models/user.rb

```diff
@@ -1,7 +1,31 @@
+require 'bcrypt'
+
 class User < ActiveRecord::Base
+
+  include BCrypt
+
   has_many :votes
   has_many :questions
   has_many :answers
-  has_many :comments, foreign_key: :author_id
+  has_many :comments, through: :answers, source: :author_id
```

Comment:
I think this doesn't need the `through` here -- a user has many comments that they have written, it's not connected via the answers table.